tools/x-repo-types/xrt_test.go

```diff
@@ -0,0 +1,104 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	name                                       string
+	xPkg, xBranch, xType, yPkg, yBranch, yType string
+	skip                                       bool
+	skipReason                                 string
+}
+
+func TestCrossRepoTypes(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	testCases := []testCase{
+		{
+			name:    "SDK: StateDelta",
+			xPkg:    "github.com/algorand/go-algorand/ledger/ledgercore",
+			xBranch: "",
+			xType:   "StateDelta",
+			yPkg:    "github.com/algorand/go-algorand-sdk/v2/types",
+			yBranch: "develop",
+			yType:   "LedgerStateDelta",
+		},
+		{
+			name:       "goal-v-sdk-genesis",
+			xPkg:       "github.com/algorand/go-algorand/data/bookkeeping",
+			xType:      "Genesis",
+			yPkg:       "github.com/algorand/go-algorand-sdk/v2/types",
+			yBranch:    "develop",
+			yType:      "Genesis",
+			skip:       true,
+			skipReason: `LEVEL 3 of goal basics.AccountData has 12 fields missing from SDK types.Account`,
+		},
+		{
+			name:       "goal-v-sdk-block",
+			xPkg:       "github.com/algorand/go-algorand/data/bookkeeping",
+			xType:      "Block",
+			yPkg:       "github.com/algorand/go-algorand-sdk/v2/types",
+			yBranch:    "develop",
+			yType:      "Block",
+			skip:       true,
+			skipReason: `Several issues. For example: LEVEL 5 of goal bookkeeping.Block is EvalDelta with field [SharedAccts](codec:"sa,allocbound=config.MaxEvalDeltaAccounts") VS SDK types.EvalDelta is missing SharedAccts field`,
+		},
+		{
+			name:    "goal-v-sdk-blockheader",
+			xPkg:    "github.com/algorand/go-algorand/data/bookkeeping",
+			xType:   "BlockHeader",
+			yPkg:    "github.com/algorand/go-algorand-sdk/v2/types",
+			yBranch: "develop",
+			yType:   "BlockHeader",
+		},
+		{
+			name:    "goal-v-sdk-stateproof",
+			xPkg:    "github.com/algorand/go-algorand/crypto/stateproof",
+			xType:   "StateProof",
+			yPkg:    "github.com/algorand/go-algorand-sdk/v2/types",
+			yBranch: "develop",
+			yType:   "StateProof",
+		},
+		{
+			name:  "goal-v-spv-stateproof",
+			xPkg:  "github.com/algorand/go-algorand/crypto/stateproof",
+			xType: "StateProof",
+			yPkg:  "github.com/algorand/go-stateproof-verification/stateproof",
+			yType: "StateProof",
+		},
+	}
+
+	for _, tc := range testCases {
+		// These should be run in serial as they modify go.mod, go.sum and typeAnalyzer/main.go
+		// TODO: it probably is preferrable to setup and `go get` everything _before_ running the tests
+		// and tearDown after the tests are done.
+		// TODO: no need to `go get`` go-algorand!!!!
```

Comment:
Reviewers: any thoughts/concerns about these?

Reply:
I've just run the following experiment locally:

1. Don't skip any tests
2. issue `go test -v -run ^TestCrossRepoTypes$ github.com/algorand/go-algorand/tools/x-repo-types`

I was observing the `git status` during the run. In the middle of a test case, you could see `go.mod` and `go.sum` being modified, but at the end of every case, the changes reverted. 

At the end of all the tests, there were no changes or other detritus remaining.

Reply:
The go.mod changes are now isolated to the sub-module. I don't see how to avoid such a side effect given the nature of the test.